cmplibrary/src/main/java/com/sourcepoint/cmplibrary/data/Service.kt

```diff
@@ -39,7 +39,8 @@ internal interface Service : NetworkClient, CampaignManager {
         messageReq: MessagesParamReq,
         pSuccess: (MessagesResp) -> Unit,
         showConsent: () -> Unit,
-        pError: (Throwable) -> Unit
+        pError: (Throwable) -> Unit,
+        onErrorFromPvData: (Throwable, Boolean) -> Unit,
```

Comment:
What's the advantage of adding a new error callback for `pv-data`? 
Could we simply _not_ call `pError` on `pv-data` and send a call to `custom-metrics` endpoint?

Reply:
@bohdan-go-wombat I might be wrong and taking things out of context, but...
 I would avoid adding new callbacks since it is a major change in a public API. 
 Can we map `PvData` error to the existing `onError` callback?

Reply:
@andresilveirah there are 2 ways of implementing this, one of which the way I implemented it, and the other one is to utilize the already existing `pError` callback with a slight modification, but this option brings a slight confusion, at least for me.

Regarding your suggestion, there are a couple of cons to this:
1. we can't just call a `custom-metrics` endpoint, since we also have a flow to send stored consent to a client via `ConsentManager` and set message error action via `ClientEventManager` in the `pError` callback, so we have to make sure that works as intended as well
2. in order to do so, we would need to add a couple of parameters to a constructor of the `Service` class, and deal with the instances inside this class, which, at least for me, seemed like an overkill

@Nevazhnovu as far as I'm concerned `Service` interface and `ServiceImpl` class are not a part of the public API (if by "public API" you mean the code that will be used by our users directly and not via our SDK API). The `Service` object is being used to create `SpConsentLib` instance, which **is** a part of the public API.

Reply:
Gotcha, makes sense.